lib_eio_linux/eio_linux.ml

```diff
@@ -286,11 +286,73 @@ let net = object
   method getnameinfo = Eio_unix.getnameinfo
 end
 
+module Process = Low_level.Process
+
+let unix_status_to_stats = function
+  | Unix.WEXITED i -> Eio.Process.Exited i
+  | Unix.WSIGNALED i -> Eio.Process.Signaled i
+  | Unix.WSTOPPED i -> Eio.Process.Stopped i
+
+let process proc : Eio.Process.t = object
+  method pid = Process.pid proc
+  method status = 
+    let status = Eio.Promise.await @@ Process.exit_status proc in
+    unix_status_to_stats status
+  method signal i = Process.signal proc i
+end
+
+let read_of_fd ~sw t =
+  match get_fd_opt t with
+  | None ->
+    let r, w = Low_level.pipe ~sw in
+    Some (flow w), r
+  | Some fd -> None, fd
+
+let write_of_fd ~sw t =
+  match get_fd_opt t with
+  | None ->
+    let r, w = Low_level.pipe ~sw in
+    Some (flow r), w
+  | Some fd -> None, fd
+
+let process_mgr = object
+  inherit Eio.Process.mgr
+  
+  method spawn ~sw ?cwd ~(stdin : #Eio.Flow.source) ~(stdout : #Eio.Flow.sink) ~(stderr : #Eio.Flow.sink) prog args = 
+    let chdir = Option.to_list cwd |> List.map (fun (_, s) -> Process.Fork_action.chdir s) in
```

Comment:
You can't ignore the base directory here (`s` is relative to that). It needs to use `fchdir` in that case.

Reply:
Hmm, I've given it a quick go in https://github.com/ocaml-multicore/eio/pull/473/commits/9f2a5ed9e89739a243e0518817b16e560278d486 which seems to be working but wasn't sure if that's right (or the cleanest way) to do it